### PR TITLE
[Docs] Update FAQ

### LINKED ISSUE #1520: [Docs] Add installation question/fix on Windows in FAQ
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the FAQ documentation but cannot get the expected help.

User may found the following error during the mmcv-installation on Windows if the version of pytorch less than 1.7.0, 
`"torch::jit::ProfileOptionalOp::Kind" may not be initialized`
mmcvv1.3.17_cuda102_py37_torch1.7.0
```
[2021-11-23T05:04:25.044Z] C:/Users/user/miniconda3/envs/mmcvv1.3.17_cuda102_py37_torch1.7.0/lib/site-packages/torch/include\torch/csrc/jit/ir/ir.h(1347): error: member "torch::jit::ProfileOptionalOp::Kind" may not be initialized
[2021-11-23T05:04:25.044Z] 
[2021-11-23T05:04:25.044Z] 1 error detected in the compilation of "C:/Users/user/AppData/Local/Temp/tmpxft_00001714_00000000-14_correlation_cuda.compute_72.cpp1.ii".
```
The solve method can be:
comment the code line: `static constexpr Symbol Kind` in `torch/include\torch/csrc/jit/ir/ir.h` or just delete it

Maybe we should add this into FQA

### PR DESCRIPTION
**Pull request by zhouzaida**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add frequently asked questions to faq.md

resolves #1520 #1518

## Modification

- docs/faq.md
- docs_zh_CN/faq.md

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -3,40 +3,85 @@
...
+
+    3. If Pytorch is correctly installed, check whether MMCV is correctly installed.
+
+        ```shell
+        python -c 'import mmcv; import mmcv.ops'
+        ```
+
+        If MMCV is correctly installed, then there will be no issue of the above two commands.
+
+    4. If MMCV and Pytorch is correctly installed, you man use `ipdb`, `pdb` to set breakpoints or directly add 'print' to debug and see which part leads the segmentation fault.
```
Comment (teamwong111) [inline docs/faq.md:71]: 1. you man --> you can
2. segmentation fault --> `segmentation fault` or "segmentation fault"

File: docs/faq.md [hunk 2]
```diff
@@ -3,40 +3,85 @@
...
+
+- "libtorch_cuda_cu.so: cannot open shared object file"
+
+    `mmcv-full` depends on the share object but it can not be found. We can check whether the object exists in `~/miniconda3/envs/{environment-name}/lib/python3.7/site-packages/torch/lib` or directly re-install the PyTorch.
```
Comment (teamwong111) [inline docs/faq.md:75]: Shall we add lib path for people who use pip ?

File: docs_zh_CN/faq.md [hunk 3]
```diff
@@ -26,12 +30,48 @@
...
+
+        是否返回True。
+
+    3. 如果 `torch` 的安装是正确的，检查是否正确编译了 MMCV。
+
+        ```shell
+        python -c 'import mmcv; import mmcv.ops'
+        ```
+
+    4. 如果 MMCV 与 PyTorch 都被正确安装了，则使用 `ipdb`, `pdb` 设置断点，直接查找哪一部分的代码导致了 `segmentation fault`。
```
Comment (teamwong111) [inline docs_zh_CN/faq.md:61]: 1. Lack of translation for "or directly add 'print'"
2. \`segmentation fault\` and "segmentation fault", which is better?

File: docs/faq.md (hunk 1 above)
Comment (teamwong111) [inline docs/faq.md:71]: 1. you man --> you can
2. segmentation fault --> \`segmentation fault\` or "segmentation fault"

**Review by teamwong111 — CHANGES_REQUESTED** [review]

File: docs/faq.md (hunk 1 above)
Comment (zhouzaida) [reply docs/faq.md:71]: `segmentation fault`

File: docs_zh_CN/faq.md (hunk 3 above)
Comment (zhouzaida) [reply docs_zh_CN/faq.md:61]: `segmentation fault` maybe better

**Comment by zhouzaida** [comment]
https://mmcv--1481.org.readthedocs.build/en/1481/faq.html
https://mmcv--1481.org.readthedocs.build/zh_CN/1481/faq.html

File: docs/faq.md [hunk 4]
```diff
@@ -3,40 +3,107 @@
 We list some common troubles faced by many users and their corresponding solutions here.
 Feel free to enrich the list if you find any frequent issues and have ways to help others to solve them.
 
+### Installation
+
+- "KeyError: "MaskRCNN: 'RefineRoIHead is not in the models registry'"
+
+    The registry mechanism will be triggered only when the file where the module is located is imported.
```
Comment (ZwwWayne) [inline docs/faq.md:10]: the file where the module is located -> the file of the module

File: docs/faq.md [hunk 5]
```diff
@@ -3,40 +3,107 @@
...
+    So you need to import that file somewhere. More details can be found at https://github.com/open-mmlab/mmdetection/issues/5974.
+
 - Compatibility issue between MMCV and MMDetection; "ConvWS is already registered in conv layer"
```
Comment (ZwwWayne) [inline docs/faq.md:13]: We can put this question to the last as this is an old issue when the ops are first migrated to mmcv from mmdet

File: docs/faq.md [hunk 6]
```diff
@@ -3,40 +3,107 @@
...
+       the Pytorch version is the same as that used for compiling mmcv
+
     3. Run `python mmdet/utils/collect_env.py` to check whether PyTorch, torchvision,
-       and MMCV are built by and running on the same environment.
+       and MMCV are built by and running on the same environment
 
-- "RuntimeError: CUDA error: invalid configuration argument".
+- "RuntimeError: CUDA error: invalid configuration argument"
 
     This error may be due to your poor GPU. Try to decrease the value of [THREADS_PER_BLOCK](https://github.com/open-mmlab/mmcv/blob/cac22f8cf5a904477e3b5461b1cc36856c2793da/mmcv/ops/csrc/common_cuda_helper.hpp#L10)
```
Comment (ZwwWayne) [inline docs/faq.md:48]: your poor GPU is not friendly and clear, rephrase it precisely.

File: docs/faq.md [hunk 7]
```diff
@@ -3,40 +3,107 @@
...
+- "error: member "torch::jit::ProfileOptionalOp::Kind" may not be initialized"
+
+  If your version of PyTorch is 1.7.0 and you are building mmcv-full on Windows, you will probably encounter the error `torch/include\torch/csrc/jit/ir/ir.h(1347): error: member "torch::jit::ProfileOptionalOp::Kind" may not be initialized`. The way to solve the error needs to modify several local files of PyTorch:
+
+  - delete `static constexpr Symbol Kind = ::c10::prim::profile;` and `tatic constexpr Symbol Kind = ::c10::prim::profile_optional;` at `torch/include\torch/csrc/jit/ir/ir.h`
+  - replace `explicit operator type&() { return *(this->value); }` with `explicit operator type&() { return *((type*)this->value); }` at `torch\include\pybind11\cast.h`
+  - replace all the `CONSTEXPR_EXCEPT_WIN_CUDA` with `const` at `torch/include\torch/csrc/jit/api/module.h`
+
+  More details can be found at https://github.com/pytorch/pytorch/pull/45956.
+### Usage
```
Comment (ZwwWayne) [inline docs/faq.md:99]: empty line above.

File: docs/faq.md [hunk 8]
```diff
@@ -3,40 +3,107 @@
 We list some common troubles faced by many users and their corresponding solutions here.
 Feel free to enrich the list if you find any frequent issues and have ways to help others to solve them.
 
+### Installation
```
Comment (ZwwWayne) [inline docs/faq.md:6]: We should re-ordering the questions based on their frequencies and their raised date.

**Review by teamwong111 — COMMENTED** [review]
No problem with the content, but my suggestion is to make every line in the zh_CN version consistent with that in the English version. For example:
```
Run `python mmdet/utils/collect_env.py` to check whether PyTorch, torchvision,
and MMCV are built for the correct GPU architecture.
``` 
Modify to
```
Run `python mmdet/utils/collect_env.py` to check whether PyTorch, torchvision and MMCV are built for the correct GPU architecture.
```

**Review by teamwong111 — APPROVED** [review]

File: docs/faq.md [hunk 9]
```diff
@@ -3,40 +3,107 @@
...
 
-- "RuntimeError: nms is not compiled with GPU support".
+- "RuntimeError: nms is not compiled with GPU support"
 
     This error is because your CUDA environment is not installed correctly.
     You may try to re-install your CUDA environment and then delete the build/ folder before re-compile mmcv.
+
+- "Segmentation fault"
+
+    1. Check you GCC version and use GCC 5.4. This usually caused by the incompatibility between PyTorch and the environment (e.g., GCC < 4.9 for PyTorch). We also recommend the users to avoid using GCC 5.5 because many feedbacks report that GCC 5.5 will cause "segmentation fault" and simply changing it to GCC 5.4 could solve the problem
```
[the remark this answers is not in the document]
Comment (zhouzaida) [reply docs/faq.md:58]: GCC 7.4 also works well.

File: docs/faq.md (hunk 8 above)
Comment (zhouzaida) [reply docs/faq.md:6]: yet

**Pull request merged**